apps/api-extractor/package.json

```diff
@@ -48,12 +48,12 @@
     "resolve": "~1.22.1",
     "semver": "~7.3.0",
     "source-map": "~0.6.1",
-    "typescript": "~4.8.4"
+    "typescript": "~5.0.4"
```

Comment:
I read the release notes for TypeScript 5.0. Thankfully, in this release none of the new features seem likely to impact API Extractor. 🎉

Some changes that potentially might cause conflicts for our tools:

- **tsconfig.json** `extends` now supports multiple base files
- `"moduleResolution": "bundler"` is new, not sure what the implications are
- `--allowArbitraryExtensions` allows arbitrary files to be referenced via `import`. API Extractor processes .d.ts files. Does this get normalized there? Or are the resulting .d.ts files only parsable by a consuming project whose `tsc` command line also specifies `--allowArbitraryExtensions` ?
- `export type *` might not be handled correctly, but if so, the fix should be easy
- Some compiler APIs and internals have changed, summarized here: https://github.com/microsoft/TypeScript/wiki/API-Breaking-Changes
- TypeScript 5.0 still supports Node 12, but we should be aware that TypeScript 5.1 will require Node 14. This will become a requirement of API Extractor, so we'd probably want to mention it in the change log or implement a version check.  I know of projects that are still using Node 12. 😆

Reply:
API Extractor doesn't like modules that are accepted by TSC by `allowArbitraryExtensions`, but I think we should fix that in a separate PR.

Reply:
Added a test for `export type *`.

Reply:
Multi-`extends` doesn't work, but that's an issue with `heft-config-file`. That'll take some more work to fix, which I think we should fix separately.